### PR TITLE
Metering as a runtime option

### LINKED ISSUE #181: Move metering contract to a runtime option

### PR DESCRIPTION
**Pull request by jakelang**
Fixes #181 
needs ci to be fixed for tests to run #193

### CODE REVIEW

File: src/hera.cpp [hunk 1]
```diff
@@ -53,6 +53,9 @@ struct hera_instance : evmc_instance {
   bool use_evm2wasm_js = false;
   bool use_evm2wasm_js_trace = false;
 #endif
+#if HERA_METERING_CONTRACT
```
Comment (axic) [inline src/hera.cpp]: IIRC my idea in the issue was to remove the `ifdef`s entirely and just rely on this runtime option.

File: src/hera.cpp [hunk 2]
```diff
@@ -390,6 +396,12 @@ int hera_set_option(
     hera->use_evm2wasm_js_trace = strcmp(value, "true") == 0;
     return 1;
   }
+#endif
+#if HERA_METERING_CONTRACT
+  if(strcmp(name, "metering-contract") == 0) {
```
Comment (axic) [inline src/hera.cpp]: I'd just call this `metering`.

**Comment by jakelang** [comment]
@axic the option is now documented in the readme.

File: src/hera.cpp [hunk 3]
```diff
@@ -325,7 +320,8 @@ evmc_result hera_execute(
 
       if (msg->kind == EVMC_CREATE && !result.isRevert && hasWasmPreamble(result.returnValue)) {
```
Comment (axic) [inline src/hera.cpp]: I'd just add the `hera->metering` check here to simplify the code (and avoid a copy).

Comment (axic) [reply src/hera.cpp]: Perhaps its good to keep the `assert > 5`, so in that case must use `move` on `returnValue`.

**Comment by axic** [comment]
Changed it myself.

**Pull request merged**